apps/system/test_softlockup/test_softlockup.c

```diff
@@ -0,0 +1,189 @@
+
+#include <stdio.h>
+#include <unistd.h>
+#include <tinyara/config.h>
+#include <pthread.h>
+#include <tinyara/wqueue.h>
+
+#ifdef CONFIG_SYSTEM_SOFTLOCKUP
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: thread_hogging_function
+ *
+ * Description:
+ *   This functions starts the infinte loop to create the hogging scenario.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+static void *thread_hogging_function(void *parameter)
+{
+	for (;;) {
+		/* Do Nothing and Hog */
+	}
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: thread_function
+ *
+ * Description:
+ *   This function terminates after a finite period.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+static void *thread_function(void *parameter)
+{
+	int i;
+	for (i = 0; i < 5; i++) {
+		sleep(1);
+	}
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: test_low_prio_hog_thread
+ *
+ * Description:
+ *   This function creates two pthreads. One pthread is of high priority and
+ *   terminates after a finite period and second pthread runs in an infinite
+ *   loop.
+ *
+ * Input parameters:
+ *   void
+ *
+ * Returned Value:
+ *   void
+ *
+ ****************************************************************************/
+
+void test_low_prio_hog_thread(void)
+{
+	pthread_t thread1;
+	pthread_t thread2;
+	pthread_attr_t attr1;
+	pthread_attr_t attr2;
+	struct sched_param sparam;
+
+	(void)pthread_attr_init(&attr1);
+	sparam.sched_priority = 225;
+	(void)pthread_attr_setschedparam(&attr1, & sparam);
+	(void)pthread_create(&thread1, &attr1, thread_function, NULL);
+	pthread_setname_np(thread1, "hp_thread");
+
+	(void)pthread_attr_init(&attr2);
+	sparam.sched_priority = 150;
+	(void)pthread_attr_setschedparam(&attr2, & sparam);
+	(void)pthread_create(&thread2, &attr2, thread_hogging_function, NULL);
+	pthread_setname_np(thread2, "lp_hog_thread");
+
```

Comment:
You should call pthead_join or detach to prevent leak.

Reply:
used pthread_detach